### PR TITLE
[Battery_plus] Fix: batteryState always return unknown on API < 26

### PR DESCRIPTION
**Pull request by Anas35**
## Description

Fixes battery state always return unknown on API lesser than 26. I manually verified by running locally. but, how do you write a test for this fix?

## Related Issues

#944 

## Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated the version in `pubspec.yaml` and `CHANGELOG.md`.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [x] No, this is *not* a breaking change.

### CODE REVIEW

**Comment by miquelbeltran** [comment]
I'd say the way to have an integration test is to create a specific emulator running a lower Android version, but IMO it won't be necessary.

File: packages/battery_plus/battery_plus/android/src/main/kotlin/dev/fluttercommunity/plus/battery/BatteryPlusPlugin.kt
Comment (miquelbeltran) [inline packages/battery_plus/battery_plus/android/src/main/kotlin/dev/fluttercommunity/plus/battery/BatteryPlusPlugin.kt:96]: I'm a bit afraid of the `!!` here, could you make it a null conditional instead?

Comment (Anas35) [reply packages/battery_plus/battery_plus/android/src/main/kotlin/dev/fluttercommunity/plus/battery/BatteryPlusPlugin.kt]: done

**Review by miquelbeltran — APPROVED** [review]

**Comment by miquelbeltran** [comment]
Thank you! I will just check this locally one more time before merging

**Review by miquelbeltran — CHANGES_REQUESTED** [review]
I noticed the method `onListen` will fail in android <26, I'd suggest to fix that before merging as well

**Review by miquelbeltran — APPROVED** [review]
Alright, I did a couple of small tweaks and verified the functionality in an emulator with API 25. All good now. Will merge once CI finishes.

**Comment by Anas35** [comment]
Thanks for the tweaks, #1119 can also be close when this merge?

**Comment by miquelbeltran** [comment]
> Thanks for the tweaks, #1119 can also be close when this merge?

yes, I will do that

**Comment by miquelbeltran** [comment]
I should have submitted that into a separated PR to earn some hacktoberfest points 😂

File: packages/battery_plus/battery_plus/android/src/main/kotlin/dev/fluttercommunity/plus/battery/BatteryPlusPlugin.kt
Comment (vbuberen) [inline packages/battery_plus/battery_plus/android/src/main/kotlin/dev/fluttercommunity/plus/battery/BatteryPlusPlugin.kt:80]: I doubt that it can be null here, but Ok.

File: packages/battery_plus/battery_plus/android/src/main/kotlin/dev/fluttercommunity/plus/battery/BatteryPlusPlugin.kt
Comment (woprandi) [inline packages/battery_plus/battery_plus/android/src/main/kotlin/dev/fluttercommunity/plus/battery/BatteryPlusPlugin.kt:95]: Why use a different way for API > O ?

**Pull request merged**
